pkg/uploadserver/uploadserver.go

```diff
@@ -422,6 +459,17 @@ func newUploadStreamProcessor(stream io.ReadCloser, dest, imageSize string, file
 	return processor.PreallocationApplied(), err
 }
 
+func newUploadArchiveStreamProcessor(stream io.ReadCloser) error {
+	destDir := common.ImporterVolumePath
+	if err := importer.CleanDir(destDir); err != nil {
+		return errors.Wrapf(err, "error removing contents of %s", destDir)
+	}
+	if err := util.UnArchiveTar(stream, destDir); err != nil {
```

Comment:
@awels should we be going to the data processor api here?  Is there currently a path that will handle this?

Reply:
That is a good question, had we gone with a single endpoint, it would have gone into an UploadDataSource, and we could have modified the Transfer function to check for content type and call untar in there just like we do in [http-datasource ](https://github.com/kubevirt/containerized-data-importer/blob/main/pkg/importer/http-datasource.go#L170-L176) But because we have 2 endpoints, I am not sure if it makes sense to duplicate it all for a new endpoint.